shared-bindings/storage/__init__.c

```diff
@@ -160,16 +161,36 @@ MP_DEFINE_CONST_FUN_OBJ_1(storage_getmount_obj, storage_getmount);
 //|     This function can be called from the REPL when ``CIRCUITPY``
 //|     has become corrupted.
 //|
+//|     :param bool extended: On boards that support ``CIRCUITPY_DUALBANK``
+//|         and ``CIRCUITPY_STORAGE_EXTEND``, the ``CIRCUITPY`` storage can be
+//|         extended by setting this to `True`. If this isn't provided or set to
+//|         `None` (default), exisiting configuration will be used.
+//|
 //|     .. warning:: All the data on ``CIRCUITPY`` will be lost, and
 //|          CircuitPython will restart on certain boards."""
 //|     ...
 //|
 
-STATIC mp_obj_t storage_erase_filesystem(void) {
-    common_hal_storage_erase_filesystem();
+STATIC mp_obj_t storage_erase_filesystem(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    enum { ARG_extended };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_extended, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+    };
+
+    #if CIRCUITPY_STORAGE_EXTEND
+    // parse args
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+    // set extended
+    bool extended = (args[ARG_extended].u_obj == mp_const_none) ? supervisor_flash_get_extended() : mp_obj_is_true(args[ARG_extended].u_obj);
+    common_hal_storage_erase_filesystem(extended);
+    #else
+    common_hal_storage_erase_filesystem(false);
```

Comment:
I think it would be good to check for the argument being other than ``None`` if CIRCUITPY_STORAGE_EXTEND is not 1, and to throw a ValueError in that case, to make it clear it isn't supported.

Reply:
I was tempted to make the new parameter not even be in the Python API if it's not supported, but maybe that's too heretical. It'd be nice not to give up those bytes of flash.

Reply:
I see what you are saying about flash size, yes, it's another 24 bytes or whatever.

Reply:
I made one minor change here to use NotImplementedError instead of adding a new message.